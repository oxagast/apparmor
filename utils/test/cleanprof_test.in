# A simple test comment which will persist
#include <tunables/global>

#include  if  exists <tunables/nothing>

  #include if exists <tunables/global>
  include if exists <tunables/global>

		alias /foo    ->    /bar	,

@{xy}    =   y   x

  abi  <abi/4.19>    ,

  @{asdf}       =   foo           ""

$foo = false

   $bar    =    true

/usr/bin/a/simple/cleanprof/test/profile {
	# Just for the heck of it, this comment won't see the day of light
	#include <abstractions/base>

#include  if  exists <foo>
	#include if exists <abstractions/base>
	include <abstractions/base>

    capability sys_admin,
    audit capability,

    change_profile -> /bin/foo,
    change_profile,

    network inet stream,
               abi  "abi/4.20"    ,
    network stream,

	#Below rule comes from abstractions/base
	allow /usr/share/X11/locale/**  r,
	allow /home/*/** r,

    ptrace tracedby peer=/bin/strace,
    ptrace tracedby,
    unix (receive) type=dgram,

    dbus send bus=session,
    dbus send bus=session peer=(label=foo),

    profile test_child /foobar {
            /etc/child rw,
            }

    set rlimit nofile <= 256,
    set rlimit nofile <= 64,

    signal set=(hup int quit ill trap abrt)
             set=(bus,fpe,,,kill,usr1)
                      set=segv set=usr2 set=pipe set=alrm set=term set=stkflt set=chld,
    signal set=(hup int quit),

    ^foo {
            /etc/fstab r,
        capability dac_override,
        }

    ^foo, # hat declarations are obsolete and will be removed when aa-cleanprof or aa-logprof writes the profile

	mount options=(rw,suid) /c -> /3,

    hat bar {
        /etc/passwd r,
        capability sys_admin,
        }

	pivot_root oldroot=/mnt/root/old/,

	deny    owner       link       /some/thing    ->     /foo/bar    ,
	unix shutdown addr=@HypotheticalServiceDaemon,

	link subset /alpha/beta -> /tmp/**,

	allow /home/foo/bar r,
	allow /home/foo/** w,
}

/usr/bin/other/cleanprof/test/profile {
	# This one shouldn't be affected by the processing
	# However this comment will be wiped, need to change that
	allow /home/*/** rw,
	allow /home/foo/bar r,
}

/what/ever/xattr   xattrs=(   foo=bar      )
       flags=(  complain
	   ) {
 /what/ever r,
 }
