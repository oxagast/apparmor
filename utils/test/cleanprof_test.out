abi <abi/4.19>,

alias /foo -> /bar,

include <tunables/global>
include if exists <tunables/nothing>

@{xy} = x y
@{asdf} = "" foo

$foo = false
$bar = true

# A simple test comment which will persist


/usr/bin/a/simple/cleanprof/test/profile {
  abi "abi/4.20",

  include <abstractions/base>
  include if exists <foo>

  set rlimit nofile <= 256,

  audit capability,

  network stream,

  dbus send bus=session,

  mount options=(rw,suid) /c -> /3,

  signal set=(abrt alrm bus chld fpe hup ill int kill pipe quit segv stkflt term trap usr1 usr2),

  pivot_root oldroot=/mnt/root/old/,

  unix (receive) type=dgram,
  unix shutdown addr=@HypotheticalServiceDaemon,

  deny owner link /some/thing -> /foo/bar,

  allow /home/*/** r,
  allow /home/foo/** w,
  link subset /alpha/beta -> /tmp/**,

  change_profile,


  hat bar {
    capability sys_admin,

    /etc/passwd r,

  }

  ^foo {
    capability dac_override,

    /etc/fstab r,

  }

  profile test_child /foobar {
    /etc/child rw,

  }
}
/usr/bin/other/cleanprof/test/profile {
  allow /home/*/** rw,
  allow /home/foo/bar r,

}
/what/ever/xattr xattrs=(   foo=bar      ) flags=(  complain ) {
  /what/ever r,

}
